packages/zwave-js/src/lib/driver/Driver.ts

```diff
@@ -2412,6 +2413,53 @@ ${handlers.length} left`,
 				if (!msg.command.moreUpdatesFollow) {
 					nodeSessions.supervision.delete(msg.command.sessionId);
 				}
+			} else if (
+				msg.command.encapsulatingCC instanceof SupervisionCCGet
+			) {
+				// check if someone is waiting for this command
+				for (const entry of this.awaitedCommands) {
+					if (entry.predicate(msg.command)) {
+						// resolve the promise - this will remove the entry from the list
+						entry.promise.resolve(msg.command);
+						return;
+					}
+				}
+
+				if (msg.command instanceof InvalidCC) {
+					const report = new SupervisionCCReport(this, {
+						sessionId: msg.command.encapsulatingCC.sessionId,
+						moreUpdatesFollow: false,
+						nodeId: msg.command.encapsulatingCC.nodeId,
+						status: SupervisionStatus.Fail,
```

Comment:
Would it make more sense to check somehow if zjs has implemented the encapsulated command and return a `NoSupport` for that, and `Fail` otherwise? I put this one down as `Fail` initially because I found that in one case the encapsulating command was `EntryControlCCNotification` which is a supported CC, but failed to parse due to a failing payload check.

Reply:
I think it would be correct to return `NoSupport` if zjs doesn't support the payload CC. Fail means that something went wrong while executing the command.
![grafik](https://user-images.githubusercontent.com/17641229/131858457-1f41146f-8bfb-4616-94b5-b90a2a4af0dd.png)
Fail means that something went wrong while executing the command:
![grafik](https://user-images.githubusercontent.com/17641229/131858581-90ca275d-0380-4236-894d-5af41098aa1c.png)

The specs don't seem to cater for the situation where the sending node sent an invalid command. I'd lean towards `NoSupport` for that case.

Reply:
I think that the names here (especially `Fail`) are misleading. Here is what I'm pretty sure the states mean:

- `NoSupport`: receiving node can't do anything, because it either doesn't understand or doesn't support the encapsulated command. The sender now knows that its command wasn't processed. Maybe it resorts to alternative actions.
- `Success`: receiving node accepted and processed the encapsulated command AND everything went exactly as the sending node would have expected. (Ex: `CurrentValue` now exactly matches the `TargetValue` in a `MultiLevelSwithcCCSet` command). The sender now knows that the receiver is in the state it expects without needing to poll for confirmation.
- `Fail`: receiving node accepted and began processing the encapsulated command, but something prevented the device from reaching the status that the sending node would have assumed. This isn't necessarily even an error. (Ex: a user might hit the "off" button during the transition of a supervised `MultiLevelSwitchCCSet` command. The receiving device aborts the supervised set and returns a `SupervisionCCReport` with `status = Fail`, because the light didn't reach the state expected by the sender). The sender knows that the receiver could be in any state, and that it needs to poll for the actual value if it cares.
- `Working`: receiving node accepted and began processing the encapsulated command. The receiving node will continue to send `SupervisionCCReports` with status updates if the sender requested them. The sender now knows that it's command was received and supported (which might be all it cares about)n and it can continue waiting for more reports if it cares about the actual final state matching expectations.

In summary:

- `NoSupport` = Error in communication of the encapsulated command. It was either nonsensical or not something the node could support. Nothing was done.
- `Fail` = Successful communication the encapsulated command. It was tried, but the result isn't what was expected. Ask to learn the actual result.
- `Success` = Successful communication of the encapsulated command. It was processed and did exactly what was expected. The assumptions on the final result match reality.
- `Working` = Successful communication of the encapsulated command. Processing is underway. We don't yet know if the assumptions on the final result will match reality.

When an S2 secured node is using `SupervisionCCGet` to send commands to the controller, it's just looking for confirm that the command was received and processed in a timely fashion, so it can continue sending commands without worrying if it needs to resync the rolling code. `Success` and `NoSupport` are the only statuses that are really possible when the `encapsulatedCC` is a notification. I don't know if there are any commands that a node can send to a controller that would `Fail` by the supervision cc meaning of `Fail`.

Reply:
> I don't know if there are any commands that a node can send to a controller that would Fail by the supervision cc meaning of Fail.

Maybe a `Scene Activation CC` which could be used to control other nodes via the controller. But that would require someone to configure the scenes on the controller, which doesn't really make sense if you have a full-blown home automation application running there.

Reply:
Makes total sense to me - `Fail` shouldn't be just a generic "something went wrong" but rather "the command itself went wrong during execution". I will update that and implement the rest of the return values.

Reply:
So, I'm not quite sure the best way to handle a failure. The `node.handleCommand` method returns void, so I can't check for any output from there. Would wrapping the call for `node.handleCommand` in a `try/catch` do the trick? I don't really see anywhere in there which throws so I'm just throwing ideas out there...

I'm also not sure on when we should send back a `Working` report, the spec says we should only do that when `Status Updates` is set to `1`, which we can do, but then I wouldn't know what to set the `Duration` to in the `Working` report - what are your thoughts on this?

Reply:
I think we can assume for now that there is no failure unless a command cannot be decoded (`InvalidCC`)